src/selfplay/loop.cc

```diff
@@ -938,14 +941,24 @@ void ProcessFile(const std::string& file, SyzygyTablebase* tablebase,
           // A blunder is defined by the played move being worse than the
           // best move by a defined threshold, missing a forced win, or
           // playing into a proven loss without being forced.
-          if ((cur.best_q - cur.played_q >
-              deblunderQBlunderThreshold) ||
-              (cur.best_q > -1 && cur.played_q < 1 &&
-               ((cur.best_q == 1 && ((cur.invariance_info & 8) != 0))
-                || cur.played_q == -1))) {
-            activeZ[0] = cur.best_q;
-            activeZ[1] = cur.best_d;
-            activeZ[2] = cur.best_m;
+          bool deblunderTriggerThreshold = (cur.best_q - cur.played_q >
+                deblunderQBlunderThreshold - deblunderQBlunderWidth / 2.0);
+          bool deblunderTriggerTerminal = (cur.best_q > -1 && cur.played_q < 1
+                && ((cur.best_q == 1 && ((cur.invariance_info & 8) != 0))
+                 || cur.played_q == -1));
+          if (deblunderTriggerThreshold || deblunderTriggerTerminal) {
+            float newZRatio = 1.0f;
+            // If width > 0 and the deblunder didn't involve a terminal
+            // position, we apply a soft threshold by averaging old and new Z.
+            if (deblunderQBlunderWidth > 0 && !deblunderTriggerTerminal) {
+              newZRatio = std::min(1.0f, (cur.best_q - cur.played_q -
```

Comment:
This file needs formatting applied -  indentations aren't consistent

Reply:
I applied clang format in Visual Studio 2019 now, thanks.